python/katana/plugin.py

```diff
@@ -71,8 +71,8 @@ def load_module(self, fullname):
                 raise ImportError(
                     f"module {fullname} already exists while trying to insert {aliased_name} at that name."
                 )
```

Comment:
side note: this confuses me, if we are trying to load the module, we first check whether it's in sys.modules. If it's not, we check that it's in the insert table and we're at the stage where the aliased_name isn't in sys.modules either, so we now import that aliased_name. Now isn't it possible for `fullname` to be in `sys.modules`? Why would that be an error?

Reply:
Hmm. Interesting. Let me look at this. I think I had the idea that a re-entrant call was possible so I wanted to check from it.

Reply:
Yeah I guess your intuition was right because after this else case we also set `module.__name__` to be `fullname` ? That's what confused me

Reply:
I think this check is insanely unlikely to actually fail. But I also think it is possible. The case I see if that the importer finds the "fullname" initially, but then during loading of the aliased package it finds a different version of the "fullname" package and loads it, so we now have two packages that want to have the same name and are loaded from different places. I think it's possible. But very very unlikely. But it doesn't hurt to check for it.